.github/workflows/buildandtest.yml

```diff
@@ -29,7 +27,24 @@ jobs:
         AZURE_TENANT: ${{ secrets.AZURE_TENANT }}
         AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
         AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
-    - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@v5
+
+  build-windows:
+    runs-on: windows-latest
```

Comment:
if implementation is the same, you can use a matrix, instead of copy-pasting the job

```
runs-on: ${{ matrix.os }}
        strategy:
            matrix:
                os: [ubuntu-latest, windows-latest]
```